sdw_updater/UpdaterApp.py

```diff
@@ -222,39 +222,30 @@ def __init__(self):
         QThread.__init__(self)
 
     def run(self):
+        results = {}
 
         # Update dom0 first, then apply dom0 state. If full state run
         # is required, the dom0 state will drop a flag.
         self.progress_signal.emit(5)
-        upgrade_generator = Updater.apply_updates(vms=["dom0"], progress_start=5, progress_end=10)
-
-        results = {}
-        for vm, progress, result in upgrade_generator:
-            results[vm] = result
-            self.progress_signal.emit(progress)
+        results["dom0"] = Updater.apply_updates_dom0()
 
         # apply dom0 state
         self.progress_signal.emit(10)
         # add to results dict, if it fails it will show error message
         results["apply_dom0"] = Updater.apply_dom0_state()
 
         self.progress_signal.emit(15)
-        # rerun full config if dom0 checks determined it's required,
-        # otherwise proceed with per-VM package updates
+        # rerun full config if dom0 checks determined it's required
         if Updater.migration_is_required():
             # Progress bar will freeze for ~15m during full state run
             self.progress_signal.emit(35)
             # add to results dict, if it fails it will show error message
             results["apply_all"] = Updater.run_full_install()
-            self.progress_signal.emit(75)
-        else:
-            upgrade_generator = Updater.apply_updates(progress_start=15, progress_end=75)
-            for vm, progress, result in upgrade_generator:
-                results[vm] = result
-                self.progress_signal.emit(progress)
+            self.progress_signal.emit(60)
 
-        # reboot vms
-        Updater.shutdown_and_start_vms()
+        # Update all VMs regardless of migrations
+        results["templates"] = Updater.apply_updates_templates()
+        self.progress_signal.emit(75)
```

Comment:
This means that the updates jump from 15% to 75% on systems where a migration is required, and they stay at 15% for quite a long time (more than ten minutes in my case).  It would be nice if we could inform the user of more of what's happening in the middle, or, if it's not possible, perhaps we decide to opt for a spinner, since the progress bar doesn't give a lot of information.